### ISSUE TITLE
Idea: Omit identical (or nearly identical) hints

### ISSUE THREAD

**Issue by blackketter** [dexterleng/vimac#357]
I often see multiple hints that appear to have the same exact center.  I wonder if it makes sense to suppress hints that have the same x,y coordinates.   

It may be that they are very slightly different (by one or two pixels) but effectively these are indistinguishable to the eye and to the mouse.

Culling these from the list might improve performance and reduce the number of 3 or 4 character hints.

Personally, I haven't ever needed to rotate through hints to get to be able to click on the right place.  Not sure if this is an issue for other users.

**Comment by dexterleng** [comment]
Currently, hints are shown in the center of the element they represent. Hence, it definitely makes sense to omit hints of the same coordinates, since the click will be performed at the same location and the effect is the same.

I did not omit hints of the same coordinates because I might change the clicking mechanism to an action-firing mechanism (e.g. AXPress).

Filtering by same/similar coordinate doesn't seem to address the root problem - why two actionable elements occupy the same hit area. I'll have to look into this root issue first, and fix identified issues with the current "is element actionable" heuristic.

**Comment by dexterleng** [comment]
> It may be that they are very slightly different (by one or two pixels) but effectively these are indistinguishable to the eye and to the mouse.

Assuming the heuristic for determining whether an element is "actionable" is correct, if any two actionable elements intersect, a hint can just be generated for one of them, since it doesn't make sense for two clickable buttons to share the same hit area.

**Comment by blackketter** [comment]
I wonder if there should be a way to display the outline of the entire element for a hint.  Is this a mode that you have for development and do you think such a thing would be useful to end users?  I'm envisioning a translucent shading that covers the whole area for which a hint is assigned.

**Comment by dexterleng** [comment]
>  Is this a mode that you have for development and do you think such a thing would be useful to end users? 

Nope, it would be really helpful for debugging, but I haven't gotten around to writing it.

I would have to do some mockups to see how it would look, although Shortcat already has something like that:
![Screenshot 2021-03-23 at 2 20 37 PM](https://user-images.githubusercontent.com/34204380/112102252-f25d4280-8be2-11eb-92e0-3feb2fd90bce.png)

**Comment by blackketter** [comment]
Thanks. That Shortcat mode is very interesting, but I'm not sure how useful it would be for end-users. 

(BTW: I hadn't tried Shortcat before.  I much prefer Vimac. :)
